waku/factory/node_factory.nim

```diff
@@ -117,10 +117,16 @@ proc setupProtocols(
   ## Optionally include persistent message storage.
   ## No protocols are started yet.
 
+  var shardCount: uint32
```

Comment:
Not sure about this. imho `pubsubTopics.len` shall be always `>0`?

`pubsubTopics` will be deprecated soon and this will be eg a cli flag `shard-count` or `network-shards`. but that's for another PR.

Reply:
True! That was done because it made everything work better when I tested not setting anything in `pubsub-topic` and only setting `shard`.

But if we currently need to use `pubsub-topic` to list all the topics in the network, then it makes sense to add the logic similar to this in the next PR, the one deprecating `pubsub-topic`

Reply:
> That was done because it made everything work better when I tested not setting anything in pubsub-topic and only setting shard.

mm IMHO `pubsub-topic` has to be always set. It could be a default value (like when using clusterid=0 or 1) but it has to be set somewhere.